controllers/controllers/workloads/cforg_controller_test.go

```diff
@@ -0,0 +1,415 @@
+				fakeClient.PatchReturns(nil)
+				fakeStatusWriter.UpdateReturns(nil)
+			})
+
+			When("a subnamespace anchor does not exists", func() {
+				BeforeEach(func() {
+					subNamespaceAnchorError = k8serrors.NewNotFound(schema.GroupResource{}, "CFOrg")
+					setSubNamespaceStatus = true
+					subNamespaceAnchor.Status.State = v1alpha2.Missing
+					createError = nil
+				})
+
+				It("creates a subnamespace anchor and returns an empty result and does not return error", func() {
+					Expect(reconcileResult).To(Equal(ctrl.Result{}))
+					Expect(reconcileErr).NotTo(HaveOccurred())
+
+					// validate the inputs to Get
+					Expect(fakeClient.GetCallCount()).To(Equal(3))
+					_, namespacedName, _ := fakeClient.GetArgsForCall(0)
+					Expect(namespacedName.Namespace).To(Equal(defaultNamespace))
+					Expect(namespacedName.Name).To(Equal(cfOrgGUID))
+					_, namespacedName, _ = fakeClient.GetArgsForCall(1)
+					Expect(namespacedName.Namespace).To(Equal(defaultNamespace))
+					Expect(namespacedName.Name).To(Equal(cfOrgGUID))
+					_, namespacedName, _ = fakeClient.GetArgsForCall(2)
+					Expect(namespacedName.Name).To(Equal(cfOrgGUID))
+
+					// validate the inputs to Create
+					Expect(fakeClient.CreateCallCount()).To(Equal(1))
+					_, createArg, _ := fakeClient.CreateArgsForCall(0)
+					castAnchor, ok := createArg.(*v1alpha2.SubnamespaceAnchor)
+					Expect(ok).To(BeTrue(), "Cast to v1alpha2.SubnamespaceAnchor failed")
+					Expect(castAnchor.ObjectMeta).To(Equal(metav1.ObjectMeta{
+						Name:      cfOrgGUID,
+						Namespace: defaultNamespace,
+						Labels: map[string]string{
+							OrgNameLabel: cfOrg.Spec.Name,
+						},
+						OwnerReferences: []metav1.OwnerReference{
+							{
+								APIVersion: APIVersion,
+								Kind:       Kind,
+								Name:       cfOrg.Name,
+								UID:        cfOrg.GetUID(),
+							},
+						},
+					}))
+					Expect(castAnchor.Status).To(Equal(v1alpha2.SubnamespaceAnchorStatus{
+						State: v1alpha2.Ok,
+					}))
+
+					// validate the inputs to Patch
+					Expect(fakeClient.PatchCallCount()).To(Equal(1))
+					_, patchedHierarchy, _, _ := fakeClient.PatchArgsForCall(0)
+					castHierarchy, ok := patchedHierarchy.(*v1alpha2.HierarchyConfiguration)
+					Expect(ok).To(BeTrue(), "Cast to v1alpha2.HierarchyConfiguration failed")
+					Expect(castHierarchy.Spec.AllowCascadingDeletion).To(BeTrue())
+
+					// validate the inputs to Status.Update
+					Expect(fakeStatusWriter.UpdateCallCount()).To(Equal(2))
+					_, updatedCFOrg, _ := fakeStatusWriter.UpdateArgsForCall(1)
+					castCFOrg, ok := updatedCFOrg.(*workloadsv1alpha1.CFOrg)
+					Expect(ok).To(BeTrue(), "Cast to workloadsv1alpha1.CFOrg failed")
+					Expect(meta.IsStatusConditionTrue(castCFOrg.Status.Conditions, StatusConditionReady)).To(BeTrue(), "Status Condition "+StatusConditionReady+" was not True as expected")
+					Expect(castCFOrg.Status.GUID).To(Equal(namespace.Name))
+				})
+			})
+
+			When("a subnamespace anchor exists", func() {
```

Comment:
In order to match @georgethebeatle proposal above (the one with `Describe("Create"...)` this could be `Describe("Update", ....)`.